pkg/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,724 @@
+package telemetry_test
```

Comment:
I'm not sure these tests add a lot of value at the moment, happy to remove them too if it makes implementation faster.